File: Ontology/ObjectPropertyRange.cs
namespace Ontology
{
    public class ObjectPropertyRange:
        ObjectPropertyAxiom,
        IObjectPropertyRange
    {
        private readonly IClassExpression _range;

        public ObjectPropertyRange(
            IOntology                 ontology,
            IObjectPropertyExpression objectPropertyExpression,
            IClassExpression          range
            ) : base(
                ontology,
                objectPropertyExpression)
        {
            _range = range;
        }

        IClassExpression IObjectPropertyRange.Range => _range;
    }
}

File: Web/ClientApp/src/app/Ontology/ObjectPropertyRange.ts
import { IAxiomVisitor } from "./IAxiomVisitor";
import { IClassExpression } from "./IClassExpression";
import { IObjectPropertyRange } from "./IObjectPropertyRange";
import { IOntology } from "./IOntology";
import { IObjectPropertyExpression } from "./IPropertyExpression";
import { ObjectPropertyAxiom } from "./ObjectPropertyAxiom";

export class ObjectPropertyRange
    extends ObjectPropertyAxiom
    implements IObjectPropertyRange
{
    constructor(
        ontology                       : IOntology,
        public ObjectPropertyExpression: IObjectPropertyExpression,
        public Range                   : IClassExpression
        )
    {
        super(ontology);
    }

    Accept(
        visitor: IAxiomVisitor
        )
    {
        visitor.ObjectPropertyRange(this);
    }
}
